packages/ifpack2/src/Ifpack2_Details_Chebyshev_def.hpp

```diff
@@ -638,16 +639,35 @@ setParameters (Teuchos::ParameterList& plist)
 
   // We don't want to fill these parameters in, because they shouldn't
   // be visible to Ifpack2::Chebyshev users.
+  if (plist.isParameter ("chebyshev: kind")) {
+    chebyshevKind = plist.get<std::string> ("chebyshev: kind");
+    TEUCHOS_TEST_FOR_EXCEPTION(
+      chebyshevKind != "first" &&
+      chebyshevKind != "textbook" &&
+      chebyshevKind != "fourth" &&
+      chebyshevKind != "opt_fourth",
+      std::invalid_argument,
+      "Ifpack2::Chebyshev: Ifpack2 only supports \"first\", \"textbook\", \"fourth\", and \"opt_fourth\", for \"chebyshev: kind\".");
+  }
+
+#ifndef IFPACK2_ENABLE_DEPRECATED_CODE
```

Comment:
Maybe I'm confused. Shouldn't this be `#ifdef`?

Reply:
My bad -- I think you are right!